fuse/src/mapping.ts

```diff
@@ -209,3 +199,92 @@ export function handleWhitelistedRemoved(event: WhitelistedRemoved): void {
   citizen.save()
 }
 
+function aggregateDailyUbiFromUBIClaimed(event: UBIClaimed, dailyUbi: DailyUBI | null): void {
+  if (dailyUbi.totalUBIDistributed == null) {
+    dailyUbi.totalUBIDistributed = event.params.amount
+  } else {
+    dailyUbi.totalUBIDistributed = dailyUbi.totalUBIDistributed.plus(event.params.amount)
+  }
+
+  if (dailyUbi.totalClaims == null) {
+    dailyUbi.totalClaims = BigInt.fromI32(1)
+  } else {
+    dailyUbi.totalClaims = dailyUbi.totalClaims.plus(BigInt.fromI32(1))
+  }
+
+  let statistics = Statistics.load('statistics')
+
+  dailyUbi.uniqueClaimers = statistics.uniqueClaimers
+
+  log.info('handleUBIClaimed dailyUbi.id {}, dailyUbi.totalUBIDistributed {}', [dailyUbi.id.toString(), dailyUbi.totalUBIDistributed.toString()])
+
+  dailyUbi.save()
+}
+
+function aggregateStatisticsFromUBIClaimed(event: UBIClaimed, statistics: Statistics | null, isUniqueClaimer: boolean): void {
+  if (statistics.totalUBIDistributed == null) {
+    statistics.totalUBIDistributed = event.params.amount
+  } else {
+    statistics.totalUBIDistributed = statistics.totalUBIDistributed.plus(event.params.amount)
+  }
+
+  if (isUniqueClaimer == true) {
+    if (statistics.uniqueClaimers == null) {
+      statistics.uniqueClaimers = BigInt.fromI32(1)
+    } else {
+      statistics.uniqueClaimers = statistics.uniqueClaimers.plus(BigInt.fromI32(1))
+    }
+
+  }
+
+  if (statistics.totalClaims == null) {
+    statistics.totalClaims = BigInt.fromI32(1)
+  } else {
+    statistics.totalClaims = statistics.totalClaims.plus(BigInt.fromI32(1))
+  }
+
+  statistics.save()
+
+}
+
+function aggregateCitizenFromUBIClaimed(event: UBIClaimed, citizen: Citizen | null): void {
```

Comment:
make sure you merge code review from previous PR correctly as you refactored the code in this PR